devito/data/utils.py

```diff
@@ -331,3 +309,75 @@ def mpi_index_maps(loc_idx, shape, topology, coords, comm):
         local_si[index] = as_tuple(rnorm_index)
         it.iternext()
     return owners, send, global_si, local_si
+
+
+def flip_idx(idx, decomposition):
```

Comment:
I know this is WIP, but before landing I'd like docstring + examples here (some for the other util below), as these are fairly "delicate" functions

Reply:
Sure, will add.

Reply:
cool thanks. Remind me: is this docstring tested via CI?

Reply:
Yes, it is.